pkg/cloudprovider/cloud.go

```diff
@@ -58,11 +58,10 @@ func GetLoadBalancerName(service *api.Service) string {
 
 // TCPLoadBalancer is an abstract, pluggable interface for TCP load balancers.
 type TCPLoadBalancer interface {
-	// TCPLoadBalancerExists returns whether the specified load balancer exists.
 	// TODO: Break this up into different interfaces (LB, etc) when we have more than one type of service
-	// TODO: This should really return the details of the load balancer so we can
-	// determine if it matches the needs of a service rather than if it exists.
-	TCPLoadBalancerExists(name, region string) (bool, error)
+	// GetTCPLoadBalancer returns whether the specified load balancer exists, and
+	// if so, what its IP address or hostname is.
```

Comment:
We could make the change locally to this interface, but that leads to a cascade of other things that'll also have to be changed - CreateTCPLoadBalancer, the service controller, the API itself (since publicIPs has been hacked to include the DNS address for AWS ELBs), and kube-proxy, at the very least.

If you think it's worth our time at this point, go ahead and file an issue or reopen #5224 - although you were the one that merged the PR introducing the overlap (#5228).